Improve wording for custom values (#85)
open-telemetry/build-tools: semantic-conventions/src/tests/data/markdown/deprecated/expected.md

**[1]:** If `net.transport` is not specified, it can be assumed to be `IP.TCP` except if `http.flavor` is `QUIC`, in which case `IP.UDP` is assumed.

> `http.flavor` If one of the listed values applies, then the respective value MUST be used, otherwise a custom value MAY be used:

Reviewer: This reads quite strange. Before the change it was a "natural" sentence but now it's just appended to the name of the attribute.

How about

> The following values are defined for `http.flavor`. If one of them applies, then ...

or at least adding a colon

> `http.flavor`: If one of the values listed below applies, then the respective value MUST be used, otherwise a custom value MAY be used:

Author: makes sense, ptal

> The following values are defined for `http.flavor`. If one of them applies, then it MUST be used, otherwise a custom value MAY be used.

Reviewer 2: I think I would still like to have the name of the attribute being described stand out more.

One suggestion is to add a section to https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/common.md or a similar place to explain the concept of "open enums" (name up for debate). Then we could write "`http.flavor` is an open enum (<-- link to definition here) with the following non-exhaustive list of possible values:"

Or we could simply do:

```suggestion
`http.flavor` SHOULD be set to one of the following values:
```

And for allow_custom_values=false, we use MUST instead of should.

If you keep it like it is now with the attribute name not at the start, I suggest to at least make the attribute name bold by adding `**` around it.

Author: ptal, it now reads:

`http.flavor` has the following list of well-known values. If one of them applies, then it MUST be used, otherwise a custom value MAY be used.
